### PR TITLE
Refactoring Markdown Files

### PR DESCRIPTION
**Pull request by zsdonghao**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
As discuss with @DEKHTIARJonathan , we would like to:

- [x] Simplify the `README.md`, use the links of [docs-example](http://tensorlayer.readthedocs.io/en/latest/user/example.html) and [awesome-tensorlayer](https://github.com/tensorlayer/awesome-tensorlayer) to replace the example list.
- [x] Add `example/README.md` with the links of [docs-example](http://tensorlayer.readthedocs.io/en/latest/user/example.html) and [awesome-tensorlayer](https://github.com/tensorlayer/awesome-tensorlayer)
- [x]  Delete https://github.com/tensorlayer/home
- [x] Should we move the installation guide into the docs? ❓

### Description
Please confirm~

### CODE REVIEW

**Comment by DEKHTIARJonathan** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/tensorlayer/tensorlayer/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Note, we hard-wrap at 80 chars and use 2 spaces after the last line.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by DEKHTIARJonathan** [comment]
@zsdonghao I have cleaned and reorganized the Readme can you have a look ?
Can you please create the file: examples/Readme.md and add the information you think are relevant to it ;)

File: README.md [hunk 1]
```diff
@@ -237,121 +71,123 @@ on a Titan X Pascal GPU. Here are the training speeds of respective tasks:
...
-# Prepare data
-X_train, y_train, X_val, y_val, X_test, y_test = tl.files.load_mnist_dataset(shape=(-1,784))
+```bash
+pip install git+https://github.com/tensorlayer/tensorlayer.git
+```
 
-# Define placeholder
-x = tf.placeholder(tf.float32, shape=[None, 784], name='x')
-y_ = tf.placeholder(tf.int64, shape=[None, ], name='y_')
+## As a TensorLayer Contributor
```
Comment (lgarithm) [inline README.md]: I think this section can be moved to `CONTRIBUTING.md`

File: README.md [hunk 2]
```diff
@@ -38,320 +36,111 @@ TensorLayer is a deep learning and reinforcement learning library on top of [Ten
...
-## Design Philosophy
-
-As TensorFlow users, we have been looking for a library that can serve for various development phases. This library is easy for beginners by providing rich neural network implementations,
-examples and tutorials. Later, its APIs shall naturally allow users to leverage the powerful features of TensorFlow, exhibiting best performance in addressing real-world problems. In the end, the extra abstraction shall not compromise TensorFlow performance, and thus suit for production deployment. TensorLayer is a novel library that aims to satisfy these requirements. It has three key features:
-
-- *Simplicity* : TensorLayer lifts the low-level dataflow abstraction of TensorFlow to **high-level** layers. It also provides users with massive examples and tutorials to minimize learning barrier.
-- *Flexibility* : TensorLayer APIs are transparent: it does not mask TensorFlow from users; but leaving massive hooks that support diverse **low-level tuning**.
-- *Zero-cost Abstraction* : TensorLayer is able to achieve the **full performance** of TensorFlow.
-
-## Negligible Overhead
```
Comment (DEKHTIARJonathan) [inline README.md]: @luomai You believe that this is not appropriate to keep ? I believed it was a good "selling argument"

Comment (luomai) [reply README.md]: Okay, I will resume it then

**Comment by DEKHTIARJonathan** [comment]
All reviews have been reset @zsdonghao @lgarithm @luomai.

Please have a look to the changes, if you agree please mark it as **accepted**. Once all of us, have marked it *accepted*, @zsdonghao  you can merge ;)

**Comment by luomai** [comment]
LGTM

**Comment by DEKHTIARJonathan** [comment]
@zsdonghao we all agree to merge, I let you do it ;)

The following need to be done:
- Merge this PR
- Delete the repository: https://github.com/tensorlayer/home

**Pull request merged**